crates/wysiwyg/src/composer_model/quotes.rs

```diff
@@ -35,7 +36,19 @@ where
         let Some(wrap_result) = self.state.dom.find_nodes_to_wrap_in_block(s, e) else {
             // No nodes to be wrapped found.
             // Adding an empty Quote block with an a single ZWSP
-            self.state.dom.append_at_end_of_document(DomNode::new_quote(vec![DomNode::new_text(S::zwsp())]));
+            let range = self.state.dom.find_range(s, e);
+            let leaves: Vec<&DomLocation> = range.leaves().collect();
+            if leaves.is_empty() {
+                self.state.dom.append_at_end_of_document(DomNode::new_quote(vec![DomNode::new_zwsp()]));
```

Comment:
Perhaps not a change for this work, and appreciate that this function wasn't added in this work, but do you think that the `DomNode::new_quote()` function should do a check to see whether the first child is a zwsp and then add it if it's missing?

It seems to me like this could potentially be very easy to miss (or might require a check of the argument before you call the function if you weren't sure whether or not there was a zwsp present).

Reply:
Actually, that's such a simple solution that I feel quite dumb for not coming up with it 😅 . It's definitely worth implementing it.

Reply:
Hmm... it might not be possible, since we actually need to also modify the current selection (which is in the `ComposerState`) and this method has no access to it.

Reply:
I also tried asserting that the 1st child is always a ZWSP, but there are some points where we're creating/merging block nodes and it still doesn't have any children. Since we're probably moving away from using ZWSP sooner than later, I think it's probably not worth it to re-write all those calls.

Reply:
Sounds reasonable, agree that it would make more sense if we knew we were going to be sticking with the approach and it's only a marginal gain anyway, not having to write it manually at the moment